logger.go

```diff
@@ -0,0 +1,153 @@
+package hedera
+
+/*-
+ *
+ * Hedera Go SDK
+ *
+ * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+type LogLevel string
+
+const (
+	LoggerLevelTrace    LogLevel = "TRACE"
+	LoggerLevelDebug    LogLevel = "DEBUG"
+	LoggerLevelInfo     LogLevel = "INFO"
+	LoggerLevelWarn     LogLevel = "WARN"
+	LoggerLevelError    LogLevel = "ERROR"
+	LoggerLevelDisabled LogLevel = "DISABLED"
+)
+
+type Logger interface {
+	SetSilent(isSilent bool)
+	SetLevel(level LogLevel) Logger
+	Error(msg string, keysAndValues ...interface{})
+	Warn(msg string, keysAndValues ...interface{})
+	Info(msg string, keysAndValues ...interface{})
+	Debug(msg string, keysAndValues ...interface{})
+	Trace(msg string, keysAndValues ...interface{})
+}
+
+type DefaultLogger struct {
+	logger *zerolog.Logger
+	level  *LogLevel
```

Comment:
Do we need the log level to be a pointer for some reason?

Reply:
It is leftover from previous refactoring. Updating it.